azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -201,15 +215,44 @@ private boolean isMultiExecutorMode() {
    */
   private void refreshExecutors() {
     synchronized (activeExecutors) {
-      // TODO: rest api call to refresh executor stats
+      ExecutorService taskExecutors =
```

Comment:
As discussed, moved executor service as member of executor manager and used futures to handle different scenarios